### PR TITLE
Example based on FaceNetEncoder

### PR DESCRIPTION
**Pull request by shivam-raj**
This example uses FaceNet Encoder and demostrates how to build and use a custom Encoder based on other models using Jina. Pull Request to upload FaceNet Encoder on Jina Hub has been made.

### CODE REVIEW

**Comment by nan-wang** [comment]
@shivam-raj hi, there. It's absolutely an awesome example! Would you please adapt the codes to the latest APIs? Please refer to [pokedex-with-bit](https://github.com/jina-ai/examples/tree/master/pokedex-with-bit) for more details. The major change is about how to feed multimedia data into the flow.

File: face-db-search/app.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+@click.option('--num_docs', '-n', default=500)
+@click.option('--top_k', '-k', default=5)
+@click.option('--path', '-p', help='Specify a JPG file or directory to query', default='')
+def main(task, num_docs, top_k, path):
+    from jina.clients.python import PyClient
+    os.environ['TMP_WORKSPACE'] = get_random_ws(os.environ['TMP_DATA_DIR'])
+    data_path = os.path.join(os.environ['TMP_DATA_DIR'], 'lfw')
+    if task == 'index':
+        flow = Flow().load_config('flow-index.yml')
+        with flow.build() as fl:
```
Comment (nan-wang) [inline face-db-search/app.py:115]:
```suggestion
        with flow:
            flow.index_file(...)
```

File: face-db-search/app.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+               sys.exit(0)
+
+        read_data_fn = read_custom_data if path else read_data
+        if path:
+            data_path = path
+
+        flow = Flow().load_config('flow-query.yml')
+        with flow.build() as fl:
+            ppr = lambda x: save_topk(x, os.path.join(os.environ['TMP_DATA_DIR'], 'query_results.png'))
+            fl.search(read_data_fn(data_path, 5), callback=ppr, top_k=top_k)
```
Comment (nan-wang) [inline face-db-search/app.py:134]:
```suggestion
        with flow:
            ppr = lambda x: save_topk(x, os.path.join(os.environ['TMP_DATA_DIR'], 'query_results.png'))
            fl.search_files(...)
```

File: face-db-search/yaml/index-chunk.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+          level: chunk
+          pruned:
+            - embedding
+            - buffer
+            - blob
+            - text
+      - !KVSearchDriver
+        with:
+          level: chunk
+          executor: BasePbIndexer
```
Comment (nan-wang) [inline face-db-search/yaml/index-chunk.yml:52]:
```suggestion
# we've simplified the codes. Now you can safely delete this part.
```

File: face-db-search/yaml/index-doc.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+        with:
+          method: query
+
+    IndexRequest:
+      - !DocPruneDriver
+        with:
+          pruned: ['chunks']
+      - !DocKVIndexDriver
+        with:
+          method: add
```
Comment (nan-wang) [inline face-db-search/yaml/index-doc.yml:20]:
```suggestion

```

**Comment by shivam-raj** [comment]
@nan-wang Thanks !!
I have updated to the latest API
I have also tested the code and it works.
Thanks alot for your suggestions !

**Comment by shivam-raj** [comment]
@nan-wang 
Can you help me with something ? I have a question
The earlier API I used returned uri in callback. 
The new API doesn't . I searched the docs but somehow in my case it doesn't return uri. I added Uri2Buffer in craft-load.yml but with no success.
I ended up using the meta-info to recreate the image from bytes and then deliver it but then loose the file-name in the process though retaining the doc_id.
Am I doing something wrong ?

**Pull request merged**